proto/api_test.go

```diff
@@ -150,10 +150,10 @@ func TestCombinable(t *testing.T) {
 
 func TestSetGoErrorCopy(t *testing.T) {
 	rh := ResponseHeader{}
-	err := &Error{Message: "test123"}
-	rh.SetGoError(err)
-	err.Message = "321tset"
-	if rh.Error.Message != "test123" {
+	oErr := &Error{Message: "test123"}
+	rh.Error = oErr
+	rh.SetGoError(&testError{})
```

Comment:
No, we have no way to copy an `error` (except through weird reflection stuff, which doesn't happen in the code). If I remember correctly this tests the line `rh.Error = &Error{}` in `SetGoError()`.

At some point it read `if rh.Error == nil { rh.Error = &proto.Error{} }` and the result would be that you could take an `&Error` into a goroutine, and the main thread might try to write another error, resulting in a race. Always creating a new `&Error` solves that and I think the above code exercises that correctly.